csrc/python_frontend/fusion_definition.cpp

```diff
@@ -145,11 +146,16 @@ void FusionDefinition::finalizeSchedule(
   FusionGuard::setCurFusion(prev_fusion_);
   prev_fusion_ = nullptr;
   if (multidevice_executor_ == nullptr) {
+    // NOTE: TMA operations require 32-bit indexing and magic zero support to be
+    // disabled.
     user_sched_->executor->compileFusion(
         user_sched_->schedule.get(),
         inputs,
         user_sched_->fusion_id_,
-        user_sched_->device_id_);
+        user_sched_->device_id_,
+        (uses_tma_ops)
```

Comment:
this feels a bit hacky here.

IIUC, we should be able to traverse the fusion IR `user_sched_->schedule` looking for `LoadStoreOpType::CpAsyncBulkTensorTile` to figure out that we should have use the modified compile param?

Reply:
Maybe it's better to do that inside compileFusion instead?

Reply:
We can update `CompileParam` in `compileFusion` in a separate PR and leave this one open as an example for now.

Reply:
Sounds good. I don't have any more concerns regarding this PR. But let's chat offline and discuss how we would want to merge this stack.